thorlcr/activities/fetch/thfetchslave.cpp

```diff
@@ -541,6 +575,18 @@ class CFetchSlaveBase : public CSlaveActivity, implements IFetchHandler
             return fpos;
         }
     }
+    virtual void serializeStats(MemoryBuffer &mb)
+    {
+        std::vector<OwnedPtr<CRuntimeStatisticCollection>> subFileStats;
+        if (fetchStream)
+        {
+            fetchStream->getFileStats(stats);
+            fetchStream->getSubFileStats(subFileStats);
```

Comment:
I am not sure why this is creating an array of CRuntimeStatisticCollection's each time serializeStats is called vs getFetchStats which reuses member 'stats' ?

Reply:
subFileStats wasn't going to be used outside this function at least the moment.  I could make it a member variable.  stats is a defined in the base class and (i didn't check but) it could be updated elsewhere.  Do I understand correctly that serializeStats is called once for each activity?

Reply:
no, it's called on a timer every ~15 seconds, and a final call at end of subgraph.
So these objects should be created once in the CFetchSlaveBase class, and passed in.